geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -0,0 +1,327 @@
+
+    def eval(left: Expr, right: Expr, args: Any*)(implicit ctx: EvaluationContext): Boolean =
+      isEqual(left.eval(args: _*).asInstanceOf[T], right.eval(args: _*).asInstanceOf[T])
+  }
+
+  def buildPred[T](f: (T, T) => Boolean): (Expr, Expr) => BinaryPredicate[T] = new BinaryPredicate[T](_, _, f)
+
+  val StrEQ    = buildPred[String](_.equals(_))
+  val IntEQ    = buildPred[Int](_ == _)
+  val IntLTEQ  = buildPred[Int](_ <= _)
+  val IntLT    = buildPred[Int](_ < _)
+  val IntGTEQ  = buildPred[Int](_ >= _)
+  val IntGT    = buildPred[Int](_ > _)
+  val DEQ      = buildPred[Double](_ == _)
+  val DLTEQ    = buildPred[Double](_ <= _)
+  val DLT      = buildPred[Double](_ < _)
+  val DGTEQ    = buildPred[Double](_ >= _)
+  val DGT      = buildPred[Double](_ > _)
+  
+  case class Not(p: Predicate) extends Predicate {
+    override def eval(args: Any*)(implicit ctx: EvaluationContext): Boolean = !p.eval(args: _*)
+  }
+
+  class BinaryLogicPredicate(l: Predicate, r: Predicate, f: (Boolean, Boolean) => Boolean) extends Predicate {
+    override def eval(args: Any*)(implicit ctx: EvaluationContext): Boolean = f(l.eval(args: _*), r.eval(args: _*))
+  }
+
+  def buildBinaryLogicPredicate(f: (Boolean, Boolean) => Boolean): (Predicate, Predicate) => BinaryLogicPredicate = new BinaryLogicPredicate(_, _, f)
+
+  val And = buildBinaryLogicPredicate(_ && _)
+  val Or  = buildBinaryLogicPredicate(_ || _)
+
+  def parseTransform(s: String): Expr = parse(TransformerParser.transformExpr, s).get
+  def parsePred(s: String): Predicate = parse(TransformerParser.pred, s).get
+
+}
+
+trait TransformerFn {
+  def eval(args: Any*): Any
+}
+
+trait TransformerFunctionFactory {
+  def functions: Seq[String]
+  def build(name: String): TransformerFn
+}
+
+class StringFunctionFactory extends TransformerFunctionFactory {
+
+  override def functions: Seq[String] =
+    Seq("stripQuotes", "trim", "capitalize", "lowercase", "regexReplace", "concat", "substr", "strlen")
+
+  def build(name: String) = name match {
+    case "stripQuotes" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String].replaceAll("\"", "")
+      StringFn(f)
+
+    case "strlen" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String].length
+      StringFn(f)
+
+    case "trim" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String].trim
+      StringFn(f)
+
+    case "capitalize" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String].capitalize
+      StringFn(f)
+
+    case "lowercase" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String].toLowerCase
+      StringFn(f)
+
+    case "regexReplace" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[Regex].replaceAllIn(args(2).asInstanceOf[String], args(1).asInstanceOf[String])
+      StringFn(f)
+
+    case "concat" =>
+      val f: (Any*) => Any = args => args(0).asInstanceOf[String] + args(1).asInstanceOf[String]
+      StringFn(f)
+
+    case "substr" =>
```

Comment:
If I'm grokking this on my first read through, I'd love to see these string processing commands documented for our users.